jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -231,27 +220,35 @@ protected boolean access(long time)
                 return false;
             _newSession=false;
             long lastAccessed = _sessionData.getAccessed();
-            if (_sessionInactivityTimer != null)
-                _sessionInactivityTimer.notIdle();
             _sessionData.setAccessed(time);
             _sessionData.setLastAccessed(lastAccessed);
-           _sessionData.calcAndSetExpiry(time);
+            _sessionData.calcAndSetExpiry(time);
             if (isExpiredAt(time))
             {
                 invalidate();
                 return false;
             }
             _requests++;
+
+            //temporarily stop the idle timer
+            if (LOG.isDebugEnabled()) LOG.debug("Session {} accessed, stopping timer, active requests={}",getId(),_requests);
+            _sessionInactivityTimer.cancel(); 
```

Comment:
This is a bit of a change of semantic.  The idle timeout is cancelled and then only reset once the count of requests goes to 0.    Previously long held requests (eg long polls) would not stop a session idling out.      This is not necessarily a bad change, just one we should call out and consider.